### PR TITLE
osd: fix ScrubMap::swap() to include all ScrubMap data members

### PR DESCRIPTION
**Pull request by ronen-fr**
and make those two into full booleans instead of fields.

### CODE REVIEW

File: src/osd/osd_types.h [hunk 1]
```diff
@@ -6107,8 +6109,8 @@ struct ScrubMap {
   std::map<hobject_t,object> objects;
   eversion_t valid_through;
   eversion_t incr_since;
-  bool has_large_omap_object_errors:1;
-  bool has_omap_keys:1;
+  bool has_large_omap_object_errors;
```
Comment (athanatos) [inline src/osd/osd_types.h:6112]: Let's add initializers for these two bools as well.

File: src/osd/osd_types.h [hunk 2]
```diff
@@ -6070,6 +6070,8 @@ std::ostream& operator<<(std::ostream& out, const PushOp &op);
 
 /*
  * summarize pg contents for purposes of a scrub
+ *
+ * If members are added to ScrubMao, make sure to modify swap().
```
Comment (rzarzynski) [inline src/osd/osd_types.h:6074]: Just a typo.

```suggestion
 * If members are added to ScrubMap, make sure to modify swap().
```

File: src/osd/osd_types.h [hunk 3]
```diff
@@ -6122,6 +6124,8 @@ struct ScrubMap {
     swap(objects, r.objects);
     swap(valid_through, r.valid_through);
     swap(incr_since, r.incr_since);
+    swap(has_large_omap_object_errors, r.has_large_omap_object_errors);
+    swap(has_omap_keys, r.has_omap_keys);
```
Comment (rzarzynski) [inline src/osd/osd_types.h:6128]: ACK!

**Review by rzarzynski — APPROVED** [review]

File: src/osd/osd_types.h (hunk 2 above)
Comment (ronen-fr) [reply src/osd/osd_types.h:6074]: Thanks. Will fix.

**Review by athanatos — APPROVED** [review]

File: src/osd/osd_types.h [hunk 4]
```diff
@@ -6071,7 +6071,7 @@ std::ostream& operator<<(std::ostream& out, const PushOp &op);
 /*
  * summarize pg contents for purposes of a scrub
  *
- * If members are added to ScrubMao, make sure to modify swap().
+ * If members are added to ScrubMap, make sure to modify swap().
```
Comment (rzarzynski) [inline src/osd/osd_types.h:6074]: Oops, this landed outside the commit that introduced the comment (the first one).

Comment (ronen-fr) [reply src/osd/osd_types.h:6074]: Fixed. Thanks

**Comment by ronen-fr** [comment]
jenkins test api

**Review by rzarzynski — APPROVED** [review]

**Comment by ronen-fr** [comment]
jenkins retest this please

**Comment by rzarzynski** [comment]
jenkins test api

**Comment by rzarzynski** [comment]
jenkins test make check

**Comment by ronen-fr** [comment]
jenkins test api

**Comment by ronen-fr** [comment]
Rebasing, based on Teuthology tests in
http://pulpito.front.sepia.ceph.com/?branch=wip-rf-scrubmap-swap
(the one failure is unrelated).

**Pull request merged**